code/rdb/endofperiod.q

```diff
@@ -8,6 +8,10 @@ endofperiod:{[currp;nextp;data]
 	/-If we are the new process, exit function, do not want to close handle
 	if[.proc.starttimeUTC = max times;
 		/-Setting variables so rdb can become the active rdb for this new period
+		.rdb.rdbpartition:`long$nextp;
+		/-send message to gateways to update the rdb attributes
+		gateh:exec w from .servers.getservers[`proctype;.rdb.gatewaytypes;()!();0b;0b];
+                .async.send[0b;;(`setattributes;.proc.procname;.proc.proctype;.proc.getattributes[])] each neg[gateh];
```

Comment:
Why do we call neg on the handles here : ``` each neg[gateh] ``` ? Is ``` neg ``` not applied within the ``` .async.send ``` function?

Reply:
so the last line of .async.send is  **.[{x@y; x(::);1b};(h;tosend);0b]},**. Where h is the handle passed. Omitting the neg would mean the query is sent across the naked handles (syncronously). (I think - may well be misintrerpretting)

Reply:
ah ok